### DISCUSSION TITLE
Fix Mo' Creatures' Custom Mob Spawner spawning mobs inside blocks, causing suffocation on world gen

### DISCUSSION THREAD

**Discussion by xJon** [ACGaming/UniversalTweaks#168] [Bugfix Suggestions]
Very common and happens over other dimensions as well, so when generating new chunks you can hear the agony of horses or ghasts suffocating.

![image](https://user-images.githubusercontent.com/8945347/235457910-acce82b2-12a3-49b8-a792-807ce1f66c69.png)

![image](https://user-images.githubusercontent.com/8945347/235549020-63e4ddf4-b40c-4f5f-aec5-9765c6a9af4a.png)

**Comment by Ski-z** [comment]
This is actually a vanilla issue, not particular to that mob.

Any large mob on hilly terrain will have the same experience. If their hitbox is spawned larger than the space available, they suffocate.

**Comment by xJon** [comment]
@Ski-z I have played modpacks without Custom Mob Spawner and have never encountered such a large amount of suffocating animals on world generation like horses.

**Comment by Ski-z** [comment]
@xJon Yes, but that's likely due to environmental factors.

First, MoCreatures has more large-hitboxes than many other mobs (in particular, horizontal hitboxes suffer the most). 

Secondly, MoCreatures causes these large hitbox mobs to spawn frequently in biomes other large-hitbox mobs aren't found in; for example, horses are almost always found in flatter plains and savanna biomes in vanilla, so we hardly ever see them suffocate. But if you use Biome Tweaker (or similar) to make horses spawn in an amplified Mountains biome, you will find suffocating horses everywhere.

Many of Mo Creatures mobs are large, and the spawning algorithm includes spawning these large mobs WAY more freuqently in biomes that are really variable in height, which leads to suffocation.

One thing you can do (which I have done successfully) is to use Mobs Properties Randomness mod, and give your selected group of creatures Levitation 2 for 2 seconds. This will make it so when they are spawned into the world, they rise upwards out of the blocks and come back down on the top, instead of inside of them.

I'm telling you this because this isn't an "issue" with the Mo Creatures spawner that can be fixed directly. You'll notice that really small mobs DON'T suffocate, even MoCreatures ones. In both of your screenshots, you can clearly see that these creatures are suffocating due to the hilly environment and their large size.

**Reply by xJon** [reply]
@Ski-z I completely understand what you're saying! That being said, flying through the Nether without Custom Mob Spawner installed, and nothing will impress you, but if you have it installed, every 10 or so seconds you will hear 1-2 Ghasts dying out of suffocation damage. That is a difference I can easily replicate, so I do suspect CMS significantly affects this issue.

**Comment by ACGaming** [comment]
Closed in favor of the upcoming Mo‘ Creatures fork.
